ice/trace.py

```diff
@@ -90,22 +92,26 @@ def url(self) -> str:
         return f"{server_url()}/traces/{self.id}"
 
     def _server_and_browser(self):
-        is_running = None
-        if settings.OUGHT_ICE_AUTO_SERVER:
-            ensure_server_running()
-            is_running = True
+        # We use this lock to avoid the server threads from burying the input prompt in [settings.py].
```

Comment:
Since you referred specifically to `[Settings.__get_and_store]` in the comment on definition, I think it makes sense to do so here as well.

'server threads' is confusing. This entire method is called from a separate thread, it doesn't necessarily start the server, and the server is started in its own process whose logs are not shown.

```suggestion
        # We use this lock to prevent logging from here (which runs in a background thread)
        # from burying the input prompt in [Settings.__get_and_store].
```

Reply:
ty! d00754e5ce56d2b3892ebb1f28a1de12cf11bf53